builtin/providers/cobbler/resource_cobbler_system.go

```diff
@@ -0,0 +1,851 @@
+				Computed: true,
+			},
+
+			"profile": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"proxy": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"redhat_management_key": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"redhat_management_server": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"status": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"template_files": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"template_remote_kickstarts": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+
+			"virt_auto_boot": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"virt_file_size": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"virt_cpus": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"virt_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"virt_path": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"virt_pxe_boot": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+
+			"virt_ram": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"virt_disk_driver": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceSystemCreate(d *schema.ResourceData, meta interface{}) error {
+	systemSyncLock.Lock()
+	defer systemSyncLock.Unlock()
+
+	config := meta.(*Config)
+
+	// Create a cobblerclient.System struct
+	system := buildSystem(d)
+
+	// Attempt to create the System
+	log.Printf("[DEBUG] Cobbler System: Create Options: %#v", system)
+	newSystem, err := config.cobblerClient.CreateSystem(system)
+	if err != nil {
+		return fmt.Errorf("Cobbler System: Error Creating: %s", err)
+	}
+
+	// Build cobblerclient.Interface structs
+	interfaces := buildSystemInterfaces(d.Get("interface").(*schema.Set))
+
+	// Add each interface to the system
+	for interfaceName, interfaceInfo := range interfaces {
+		log.Printf("[DEBUG] Cobbler System Interface %#v: %#v", interfaceName, interfaceInfo)
+		if err := newSystem.CreateInterface(interfaceName, interfaceInfo); err != nil {
+			return fmt.Errorf("Cobbler System: Error adding Interface %s to %s: %s", interfaceName, newSystem.Name, err)
+		}
+	}
+
+	log.Printf("[DEBUG] Cobbler System: Created System: %#v", newSystem)
+	d.SetId(newSystem.Name)
+
+	log.Printf("[DEBUG] Cobbler System: syncing system")
```

Comment:
@mongrelion I completely agree that a final Terraform step to run the sync would work best.

I just ran the acceptance test suite with 200 systems and it passed. No doubt that if there is some other activity going on, it might interfere with the series of syncs, but in general, I think the `sync.Mutex` is pretty stable.

Thoughts?